# HG changeset patch
# User John Donoghue
# Date 1428000617 14400
#      Thu Apr 02 14:50:17 2015 -0400
# Node ID e70df1ff58b7271769e721e41424559f55062dc0
# Parent  1929d9218ac0f622f6577a2cec31552721da03cc
Fix warning from changeset 519eb032ac35

* libgui/graphics/PopupMenuControl.cc
  (PopupMenuControl::update): dont warn if value is same as current value

diff --git a/libgui/graphics/PopupMenuControl.cc b/libgui/graphics/PopupMenuControl.cc
--- a/libgui/graphics/PopupMenuControl.cc
+++ b/libgui/graphics/PopupMenuControl.cc
@@ -102,20 +102,20 @@ void PopupMenuControl::update (int pId)
       m_blockUpdate = true;
         {
           Matrix value = up.get_value ().matrix_value ();
 
           if (value.numel () > 0)
             {
               int newIndex = int (value(0)) - 1;
 
-              if (newIndex >= 0 && newIndex < box->count ()
-                  && newIndex != box->currentIndex ())
+              if (newIndex >= 0 && newIndex < box->count ())
                 {
-                  box->setCurrentIndex (newIndex);
+                  if (newIndex != box->currentIndex ())
+                    box->setCurrentIndex (newIndex);
                 }
               else
                 warning ("popupmenu value not within valid display range");
             }
         }
       m_blockUpdate = false;
       break;
 
